src/categorical_algebra/Slices.jl

```diff
@@ -0,0 +1,133 @@
+module Slices
+export Slice, SliceMorphism,SliceLimit,SliceColimit, pushout_complement
```

Comment:
Since they are exported, the structs `Slice` and `SliceMorphism` should have docstrings.

Also, my guess is that `SliceLimit` and `SliceColimit` are implementation details that should not be exported. The philosophy here is that everything one can do with a (co)limit object is encapsulated by the accessor functions (`apex`, `legs`, etc) and the universal property (`universal`).